File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let i = 0
    let j = 0
    while(j < nums.length){
        if (nums[j] > nums[i]){
            nums[i+1] = nums[j]
            i+=1
        }
        j+=1
    }
    return i + 1
};

File: remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int i = -1;
        int j = 0;
        int highest = Integer.MIN_VALUE;
        while(j<nums.length){
            if(nums[j]>highest){
                highest = nums[j];
                int temp = nums[i+1];
                nums[i+1]=nums[j];
                nums[j]=temp;
                i++;
            }
            j++;
        }
        return i+1;
    }
}